src/data_processing/wiki_to_netflix.py

```diff
@@ -0,0 +1,133 @@
+    if key == 'genreLabel':
+        return list({d['genreLabel']['value'] for d in data['results']['bindings'] if 'genreLabel' in d})
+    return data['results']['bindings'][0][key]['value'].split('/')[-1] 
+
+# Getting list of movie IDs, genre IDs, and director IDs from request
+def wiki_query(data_csv, user_agent):
+    wiki_movie_ids = []
+    wiki_genres = []
+    wiki_directors = []
+        
+    for row in data_csv:
+        if row[1] == "NULL":
+            continue
+
+        SPARQL = '''
+        SELECT * WHERE {
+            SERVICE wikibase:mwapi {
+                bd:serviceParam wikibase:api "EntitySearch" ;
+                                wikibase:endpoint "www.wikidata.org" ;
+                                mwapi:search "%(Title)s" ;
+                                mwapi:language "en" .
+                ?item wikibase:apiOutputItem mwapi:item .
+            }
+
+            ?item wdt:P31/wdt:P279* wd:Q11424 .
+            
+            {
+                # Get US release date
+                ?item p:P577 ?releaseDateStatement .
+                ?releaseDateStatement ps:P577 ?releaseDate .
+                ?releaseDateStatement pq:P291 wd:Q30 .  
+            }
+            UNION
+            {
+                # Get unspecified release date
+                ?item p:P577 ?releaseDateStatement .
+                ?releaseDateStatement ps:P577 ?releaseDate .
+                FILTER NOT EXISTS { ?releaseDateStatement pq:P291 ?country }
+            }
+        
+            FILTER (YEAR(?releaseDate) = %(Year)d) .
+
+            ?item rdfs:label ?itemLabel .
+            FILTER (lang(?itemLabel) = "en") .
+
+            OPTIONAL {
+                ?item wdt:P136 ?genre .
+                ?genre rdfs:label ?genreLabel .
+                FILTER (lang(?genreLabel) = "en") .
+            }
+
+            OPTIONAL {?item wdt:P57 ?director.
+                            ?director rdfs:label ?directorLabel.
+                            FILTER (lang(?directorLabel) = "en")}
+
+            SERVICE wikibase:label { bd:serviceParam wikibase:language "en" . }
+            }
+    
+        ''' % {'Title': row[2], 'Year': int(row[1])}
+
+        response = requests.post('https://query.wikidata.org/sparql',
+                    headers={'User-Agent': user_agent},
+                    data={
+                    'query': SPARQL,
+                    'format': 'json',
+                    }
+        )
+        response.raise_for_status() 
+        
+        data = response.json()
+        
+        wiki_movie_ids.append(wiki_feature_info(data, 'item'))
+        wiki_genres.append(wiki_feature_info(data, 'genreLabel'))
+        wiki_directors.append(wiki_feature_info(data, 'directorLabel'))
+    
+    return(wiki_movie_ids, wiki_genres, wiki_directors)
+
+# Calling all functions
+def process_data(test=False):
+    missing_count = 0
+    processed_data = []
+
+    netflix_file = read_netflix_txt(os.path.join(base_dir, 'netflix_movies.txt'), test)
```

Comment:
Maybe we should leave this as `movie_data.txt` so that we as developers can simply extract the netflix data set to `/data` and not have to rename the file?

Reply:
Yeah that makes sense. I forgot what the original name was tbh. Fixed

Reply:
Sorry, my mistake, the file is called `movie_titles.txt`.

Reply:
Updated to movie_titles.txt